apiserver/facades/client/application/application_unit_test.go

```diff
@@ -35,54 +38,124 @@ import (
 	"github.com/juju/juju/core/instance"
 	"github.com/juju/juju/core/model"
 	"github.com/juju/juju/core/network"
+	"github.com/juju/juju/core/network/firewall"
 	"github.com/juju/juju/core/status"
 	"github.com/juju/juju/environs"
+	"github.com/juju/juju/environs/config"
 	"github.com/juju/juju/rpc/params"
 	"github.com/juju/juju/state"
+	"github.com/juju/juju/state/stateenvirons"
 	"github.com/juju/juju/storage"
 	"github.com/juju/juju/storage/provider"
 	coretesting "github.com/juju/juju/testing"
 	"github.com/juju/juju/tools"
 	jujuversion "github.com/juju/juju/version"
 )
 
+type mockLXDProfilerCharm struct {
+	*mocks.MockCharm
+	*mocks.MockLXDProfiler
+}
+
 type ApplicationSuite struct {
-	testing.IsolationSuite
-	coretesting.JujuOSEnvSuite
-	backend            mockBackend
-	model              mockModel
-	leadership         mockLeadership
-	endpoints          []state.Endpoint
-	relation           mockRelation
-	application        mockApplication
-	storagePoolManager *mockStoragePoolManager
-	registry           *mockStorageRegistry
-	updateSeries       *mockUpdateSeries
-	supportedFeatures  coreassumes.FeatureSet
-
-	caasBroker   *mockCaasBroker
-	env          environs.Environ
-	blockChecker mockBlockChecker
-	authorizer   apiservertesting.FakeAuthorizer
-	api          *application.APIv14
+	testing.CleanupSuite
+
+	api *application.APIv14
+
+	backend            *mocks.MockBackend
+	storageAccess      *mocks.MockStorageInterface
+	authorizer         apiservertesting.FakeAuthorizer
+	blockChecker       *mocks.MockBlockChecker
+	model              *mocks.MockModel
+	leadershipReader   *mocks.MockReader
+	storagePoolManager *mocks.MockPoolManager
+	registry           *mocks.MockProviderRegistry
+	caasBroker         *mocks.MockCaasBrokerInterface
+
 	deployParams map[string]application.DeployApplicationParams
 }
 
 var _ = gc.Suite(&ApplicationSuite{})
 
-func (s *ApplicationSuite) setAPIUser(c *gc.C, user names.UserTag) {
-	s.authorizer.Tag = user
-	s.storagePoolManager = &mockStoragePoolManager{storageType: k8sconstants.StorageProviderType}
-	s.registry = &mockStorageRegistry{}
-	s.caasBroker = &mockCaasBroker{}
+// Alternate placing storage instaces in detached, then destroyed
+func fakeClassifyDetachedStorage(
+	_ storagecommon.VolumeAccess,
+	_ storagecommon.FilesystemAccess,
+	storage []state.StorageInstance,
+) ([]params.Entity, []params.Entity, error) {
+	destroyed := make([]params.Entity, 0)
+	detached := make([]params.Entity, 0)
+	for i, stor := range storage {
+		if i%2 == 0 {
+			detached = append(detached, params.Entity{stor.StorageTag().String()})
+		} else {
+			destroyed = append(destroyed, params.Entity{stor.StorageTag().String()})
+		}
+	}
+	return destroyed, detached, nil
+}
+
+func fakeSupportedFeaturesGetter(stateenvirons.Model, environs.NewEnvironFunc) (coreassumes.FeatureSet, error) {
+	return coreassumes.FeatureSet{}, nil
+}
+
+func (s *ApplicationSuite) SetUpTest(c *gc.C) {
+	s.CleanupSuite.SetUpTest(c)
```

Comment:
You don't need to explicitly call SetUpTest(c). It is automatically done for you by the BaseSuite which in turn is setup by the suiteRunner in gocheck once you embed CleanupSuite.

Reply:
Ah I did not know that. This is how it seems to be done in almost every other test suite